### ISSUE TITLE
set ingress.port=89 causes installation error "incompatible types for comparison"

### ISSUE THREAD

**Issue by cawoodm** [nano-byte/helm-charts#11]
```bash
$helm install nginx89 nano-byte/generic-service --set image.repository=nginx --set ingress.enabled=true --set ingress.port=89 --set image.tag=latest

Error: INSTALLATION FAILED: template: generic-service/templates/controller.yaml:161:86: executing "generic-service/templates/controller.yaml" at <ne .Values.monitoring.port .Values.ingress.port>: error calling ne: incompatible types for comparison
```

Without `--set ingress.port=89` the error does not occur.

**Comment by cawoodm** [comment]
Also, no ingress was created.

**Comment by bastianeicher** [comment]
Thanks for reporting this issue.

The problem arises because the Helm chart always expects `ingress.port` and `monitoring.port` to be integers. However, Helm interprets values set via `--set` as strings. Only when set via a `values.yaml` file a value like `89` would be interpreted as an integer.

I've added an explicit cast to `int` to fix this in d5c60efec476989f7d019a2b171495f8205f9a78 and released version 1.2.5 with the fix.

**Comment by cawoodm** [comment]
Funny that `--set` gets interpreted as a string when there is a `--set-string` parameter [docs](https://helm.sh/docs/helm/helm_upgrade/#synopsis):
> To override values in a chart, use either the '--values' flag and pass in a file or use the '--set' flag and pass configuration from the command line, to force string values, use '--set-string'.
